modules/45_carbonprice/NDC/postsolve.gms

```diff
@@ -5,65 +5,49 @@
-if(iteration.val lt 8,  
- p45_factorRescaleCO2Tax(regi) = min((( max(0.1, p45_actual_co2eq_woLU_regi(regi)/p45_ref_co2eq_woLU_regi(regi) ) )** 4),max(2-iteration.val/15,1.01-iteration.val/10000)); !! use max(0.1, ...) to make sure that negative emission values cause no problem
-elseif iteration.val lt 15,
- p45_factorRescaleCO2Tax(regi) = min((( max(0.1, p45_actual_co2eq_woLU_regi(regi)/p45_ref_co2eq_woLU_regi(regi) ) )**3),max(2-iteration.val/15,1.01-iteration.val/10000));
-elseif iteration.val lt 23,
- p45_factorRescaleCO2Tax(regi) = min((( max(0.1, p45_actual_co2eq_woLU_regi(regi)/p45_ref_co2eq_woLU_regi(regi) ) )**2),max(2-iteration.val/15,1.01-iteration.val/10000));
-else
- p45_factorRescaleCO2Tax(regi) = min((( max(0.1, p45_actual_co2eq_woLU_regi(regi)/p45_ref_co2eq_woLU_regi(regi) ) )**1),max(2-iteration.val/15,1.01-iteration.val/10000));
-); 
-
-pm_taxCO2eq(t,regi)$(t.val gt 2016 AND t.val ge cm_startyear AND t.val lt 2031) = max(1* sm_DptCO2_2_TDpGtC,pm_taxCO2eq(t,regi) * p45_factorRescaleCO2Tax(regi) );
-p45_factorRescaleCO2TaxTrack(iteration,regi) = p45_factorRescaleCO2Tax(regi);
+
+if(       iteration.val lt  8, p45_adjust_exponent = 4;
+   elseif iteration.val lt 15, p45_adjust_exponent = 3;
+   elseif iteration.val lt 23, p45_adjust_exponent = 2;
+   else                        p45_adjust_exponent = 1;
+);
+
+p45_factorRescaleCO2Tax(p45_NDC_year_set(ttot,regi)) =
+  min((( max(0.1, (p45_actual_co2eq_woLU_regi(ttot,regi)+0.0001)/(p45_ref_co2eq_woLU_regi(ttot,regi)+0.0001) ) )**p45_adjust_exponent),max(2-iteration.val/15,1.01-iteration.val/10000));
+*** use max(0.1, ...) to make sure that negative emission values cause no problem, use +0.0001 such that net zero targets cause no problem
+
+pm_taxCO2eq(t,regi)$(t.val gt 2016 AND t.val ge cm_startyear AND t.val le p45_last_NDC_year(regi)) = max(1* sm_DptCO2_2_TDpGtC,pm_taxCO2eq(t,regi) * p45_factorRescaleCO2Tax(t,regi) );
+p45_factorRescaleCO2TaxTrack(iteration,ttot,regi) = p45_factorRescaleCO2Tax(ttot,regi);
 
 display p45_factorRescaleCO2TaxTrack;
 
 ***carbon price safety valve for low-income and lower-middle income regions: max 1/4 of EUR carbon price
 ***pm_taxCO2eq(t,"IND")$(t.val gt 2014 AND t.val lt 2036) = min(pm_taxCO2eq(t,"IND"),0.25*pm_taxCO2eq(t,"EUR"));
 ***pm_taxCO2eq(t,"AFR")$(t.val gt 2014 AND t.val lt 2036) = min(pm_taxCO2eq(t,"AFR"),0.25*pm_taxCO2eq(t,"EUR"));
-****** regions with some low-income or lower-middle income regions: max 50%                                                 
+***regions with some low-income or lower-middle income regions: max 50%
 ***pm_taxCO2eq(t,"LAM")$(t.val gt 2014 AND t.val lt 2036) = min(pm_taxCO2eq(t,"LAM"),0.5*pm_taxCO2eq(t,"EUR"));
 ***pm_taxCO2eq(t,"MEA")$(t.val gt 2014 AND t.val lt 2036) = min(pm_taxCO2eq(t,"MEA"),0.5*pm_taxCO2eq(t,"EUR"));
 ***pm_taxCO2eq(t,"OAS")$(t.val gt 2014 AND t.val lt 2036) = min(pm_taxCO2eq(t,"OAS"),0.5*pm_taxCO2eq(t,"EUR"));
```

Comment:
do you still need this out-commented code?

Reply:
I don't need it.

Reply:
please, delete it

Reply:
done